### PR TITLE
validate all class/define params

### LINKED ISSUE #392: validate all DSL class/define params

### PR DESCRIPTION
**Pull request by jhoblitt**
Resolves #392

### CODE REVIEW

**Comment by jhoblitt** [comment]
This is not yet ready for merge as I'm in the process of auditing.

**Comment by jhoblitt** [comment]
This should be ready for merge now.  It is likely to break some manifests due to the new strict type checking but it should be trivial. Ie., removing quotes from around an integer value.

**Comment by jhoblitt** [comment]
I'm going to go ahead and merge this so that it has time to circulate on master for a bit before another minor release (which I hope is coming soonish).

File: manifests/augeas.pp [hunk 1]
```diff
@@ -42,35 +42,36 @@
...
+    true: {
       jenkins::plugin {$name:
         version       => $plugin_version,
         manage_config => false,
         before        => Augeas["jenkins::augeas: ${name}"],
       }
     }
-    false,'false': { # lint:ignore:quoted_booleans
-      #do nothing
+    false: {
```
Comment (PierreR) [inline manifests/augeas.pp]: I am not sure why you want to keep the false case. As an outsider to the code, it looks idiotic ;-)

Comment (jhoblitt) [inline manifests/augeas.pp]: Agreed.  Interested in a PR'ing a cleanup?

**Pull request merged**